src/transformers/trainer.py

```diff
@@ -556,6 +555,9 @@ def train(self, model_path: Optional[str] = None):
     def _log(self, logs: Dict[str, float], iterator: Optional[tqdm] = None) -> None:
         if self.epoch is not None:
             logs["epoch"] = self.epoch
+        if self.global_step is None:
+            # when logging evaluation metrics without training
+            self.global_step = 0
         if self.tb_writer:
```

Comment:
Same thing here.

Reply:
Here I'm pinging @julien-c because I initially had it in `__init__` and he preferred to move it within the training loop.
I've been trying to apply the same logic with `TFTrainer` but I have no issue in doing something different.

Reply:
We don't expect to have both trainer exactly the same. The logging part hasn't been a lot discussed yet, it is currently the main difference and I think we should take some time to properly rethink that part.